scripts/release/update_retraction_table.py

```diff
@@ -1,208 +1,243 @@
+        table_id (str): Synapse Table ID
+        condition (str): SQL condition for querying the table (optional)
     
     Returns:
-        Dataframe: synapse table
+        pd.DataFrame: DataFrame containing the queried table data
     """
-    if condition:
-        condition = " WHERE "+condition
-    synapse_table = syn.tableQuery("SELECT * from %s%s" % (table_id,condition))
-    synapse_table = synapse_table.asDataFrame()
-    return(synapse_table)
-
-def setup_custom_logger(name):
-    """Set up customer logger
+    condition_str = f" WHERE {condition}" if condition else ""
+    query = f"SELECT * FROM {table_id}{condition_str}"
+    synapse_table = syn.tableQuery(query)
+    return synapse_table.asDataFrame()
 
+def setup_logger(name, log_file='log.txt'):
+    """Set up a custom logger to log information to both console and a file.
+    
     Args:
-        name (String): Name of the logger 
+        name (str): Logger name
+        log_file (str): Path to the log file
     
     Returns:
-       logger
+        logging.Logger: Configured logger object
     """
-    formatter = logging.Formatter(fmt='%(asctime)s %(levelname)-8s %(message)s',
-                                  datefmt='%Y-%m-%d %H:%M:%S')
-    handler = logging.FileHandler('log.txt', mode='w')
-    handler.setFormatter(formatter)
-    screen_handler = logging.StreamHandler(stream=sys.stdout)
-    screen_handler.setFormatter(formatter)
+    formatter = logging.Formatter('%(asctime)s %(levelname)-8s %(message)s', datefmt='%Y-%m-%d %H:%M:%S')
+    file_handler = logging.FileHandler(log_file, mode='w')
+    file_handler.setFormatter(formatter)
+    
+    console_handler = logging.StreamHandler(stream=sys.stdout)
+    console_handler.setFormatter(formatter)
+    
     logger = logging.getLogger(name)
     logger.setLevel(logging.DEBUG)
-    logger.addHandler(handler)
-    logger.addHandler(screen_handler)
-    return(logger)
+    logger.addHandler(file_handler)
+    logger.addHandler(console_handler)
+    
+    return logger
 
 def synapse_login(synapse_config):
-    """Log into Synapse
-
+    """Log into Synapse using the provided configuration file.
+    
     Args:
-        synapse_config (String): File path to the Synapse config file
-        
+        synapse_config (str): Path to the Synapse configuration file
+    
     Returns:
-        Synapse object
+        Synapse: Synapse client object
     """
     try:
         syn = synapseclient.login(silent=True)
     except Exception:
         syn = synapseclient.Synapse(configPath=synapse_config, silent=True)
         syn.login()
-    return(syn)
+    return syn
 
 def update_version(syn, table_id, comment):
+    """Update the version of the table with a new snapshot and comment.
+    
+    Args:
+        syn (Synapse): Synapse client object
+        table_id (str): Synapse Table ID
+        comment (str): Comment to attach to the snapshot
     """
-    Update the table version with given table ID and comment
```

Comment:
I actually think there is a syn.create_snapshot function that we could look into here

Reply:
Yup, [here is an example](https://github.com/Sage-Bionetworks/genie-bpc-pipeline/blob/d05e92749706e464af9ea0f3797ffc1f318c272b/scripts/references/update_potential_phi_fields_table.R#L146-L153)